prover/crates/bin/circuit_prover/src/circuit_prover.rs

```diff
@@ -121,9 +131,11 @@ impl CircuitProver {
                 "failed to get setup data for key {setup_data_key:?}"
             ))?
             .clone();
+        let snarkify_prover_clone = self.snarkify_prover.clone();
```

Comment:
nit: Maybe use Arc<SnarkifyProver> to avoid clone?

Reply:
good call. This can avoid cloning objects with large data.